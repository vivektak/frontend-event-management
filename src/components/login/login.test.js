import React from 'react';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import Login from './login';
Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('Render Login Component', () => {

    const wrapper = shallow(<Login />)
    test('should match the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });

    test('should render component ', () => {
        const element = wrapper.find(`[data-test="login-component"]`);
        expect(element.length).toBe(1)
    });
    
    test('should render mobile input box ', () => {
        const element = wrapper.find(`[name='mobile']`);
        expect(element.length).toBe(1)
    });
    
    test('should render otp input box ', () => {
        const element = wrapper.find(`[name='otp']`);
        expect(element.length).toBe(1)
    });
    
    test('should render Button ', () => {
        const element = wrapper.find(`[variant="contained"]`);
        expect(element.length).toBe(1)
    });
    
    test('should call handleSubmit on click ', () => {
        const handleSubmit = jest.fn();
        const button = shallow((<button name='login' onClick={handleSubmit}>Login</button>));
        button.find(`[name='login']`).simulate('click');
        expect(handleSubmit).toHaveBeenCalled();
    });
    
    test('should call handleValidate on click ', () => {
        const handleValidate = jest.fn();
        const button = shallow((<button name='validate' onClick={handleValidate}>Validate</button>));
        button.find(`[name='validate']`).simulate('click');
        expect(handleValidate).toHaveBeenCalled();
    });
    
    test('should call setMobile on click ', () => {
        const setMobile = jest.fn();
        const button = shallow((<textfield name="mobile" onClick={setMobile} />));
        button.find(`[name="mobile"]`).simulate('click');
        expect(setMobile).toHaveBeenCalled();
    });
    
    test('should call setMobileError on click ', () => {
        const setMobileError = jest.fn();
        const button = shallow((<textfield name="mobile" onBlur={setMobileError} />));
        button.find(`[name="mobile"]`).simulate('blur');
        expect(setMobileError).toHaveBeenCalled();
    });
    
    test('should call setOtp on click ', () => {
        const setOtp = jest.fn();
        const button = shallow((<textfield name="otp" onClick={setOtp} />));
        button.find(`[name="otp"]`).simulate('click');
        expect(setOtp).toHaveBeenCalled();
    });
    
    test('should call setOtpError on click ', () => {
        const setOtpError = jest.fn();
        const button = shallow((<textfield name="otp" onBlur={setOtpError} />));
        button.find(`[name="otp"]`).simulate('blur');
        expect(setOtpError).toHaveBeenCalled();
    });
});

