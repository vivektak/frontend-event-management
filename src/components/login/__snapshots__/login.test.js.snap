// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login Component should match the snapshot 1`] = `"<div data-test=\\"login-component\\" class=\\"login-header\\"><div class=\\"MuiPaper-root MuiCard-root card-width MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"event-management-box\\"><h5>Event Management</h5></div><div class=\\"MuiCardContent-root\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\" for=\\"standard-name\\" id=\\"standard-name-label\\">Mobile Number</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" id=\\"standard-name\\" name=\\"mobile\\" value=\\"\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\" style=\\"display:none\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\" for=\\"standard-name\\" id=\\"standard-name-label\\">OTP</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" id=\\"standard-name\\" name=\\"otp\\" value=\\"\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div></div><div class=\\"MuiCardActions-root MuiCardActions-spacing\\" style=\\"display:flex;justify-content:center\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained button login-button MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" name=\\"login\\"><span class=\\"MuiButton-label\\">Login</span></button></div><div class=\\"notification-container notification-container-empty\\"><span></span></div></div></div>"`;
